src/ServiceControl/Operations/ErrorIngestionCustomCheck.cs

```diff
@@ -0,0 +1,40 @@
+﻿namespace ServiceControl.Operations
+{
+    using System;
+    using System.Threading.Tasks;
+    using NServiceBus.CustomChecks;
+
+    class ErrorIngestionCustomCheck : CustomCheck
+    {
+        readonly State criticalErrorHolder;
+
+        public ErrorIngestionCustomCheck(State criticalErrorHolder) 
+            : base("Failed message ingestion process", "ServiceControl", TimeSpan.FromSeconds(5))
+        {
+            this.criticalErrorHolder = criticalErrorHolder;
+        }
+
+        public override Task<CheckResult> PerformCheck()
+        {
+            var failure = criticalErrorHolder.GetLastFailure();
+            if (failure == null)
+            {
+                return Task.FromResult(CheckResult.Pass);
+            }
+
+            return Task.FromResult(CheckResult.Failed(failure));
+        }
+
+        public class State
+        {
+            volatile string lastFailure;
```

Comment:
If you make this a Task<string> you can optimize this for the customer. This would be actually an excellent case for ValueTask :(

Reply:
You mean to have a `ValueTask` instead of `string` here?

Reply:
yeah but we don't have it available so I would basically directly cache `Task<string>` and use `Task.FromResult` here